msteams-platform/apps-in-teams-meetings/build-apps-for-anonymous-user.md

```diff
@@ -0,0 +1,226 @@
+---
+title: Build apps for anonymous user
+author: v-sdhakshina
+description: In this article, learn how to test the experience delivered to the anonymous users in meeting apps with all admin settings.
+ms.topic: conceptual
+ms.author: v-sdhakshina
+ms.localizationpriority: medium
+---
+
+# Build apps for anonymous users
+
+You can test the experience delivered to anonymous users by selecting the meeting join URL present in the meeting invite and joining the meeting via private window.
+
+> [!NOTE]
+>
+> * Side panel tabs and content bubbles aren't available for anonymous users at the moment.
+> * You can build conversational extensibility solutions to engage with meeting participants.
+> * Apps shared on stage by regular in-tenant users will render for anonymous users.
+
+## Tenant admin setting for anonymous user app interaction
+
+Teams admin can use the admin portal to enable or disable anonymous user app interaction for entire tenant. This setting is enabled by default, unless overridden by the admin. You must enable this to support anonymous user interaction with apps in meetings, [Allow anonymous users to interact with apps in meetings](/microsoftteams/meeting-settings-in-teams).
+
+> [!NOTE]
+> Teams Client and Bot APIs set the tenantId field from meeting organizer tenant Id for anonymous user.
+
+## In-Meeting getContext from Teams client SDK
+
+Apps receive the following info pertaining to the anonymous user through the Teams Client SDK `getContext API` when the shared app stage is loaded. Apps can recognize the users as anonymous by the **Unknown** userLicenseType.
+
+```csharp
+"userObjectId": "8:anon:<<User GUID>>",
+"userLicenseType": "Unknown",
+"loginHint": "8:teamsvisitor:<<ID>>",
+"userPrincipalName": "8:teamsvisitor:<<ID>>",
+"tid": "<<GUID>>"
+```
+
+## Payloads of Bot API for anonymous users in a meeting chat
+
+Bot APIs make your bot meeting aware. Following are some examples of bot APIs and how they allow you to differentiate between regular and anonymous users:
+
+**Paged Get Members API**
+
+```csharp
+{ 
+  "members": [ 
+    { 
+      "id": "<<GUID>>", 
+      "name": "AnonTest (Guest)",  
+      "tenantId": "<<GUID>>", 
+      "userRole": "anonymous" 
+    } 
+  ] 
+} 
+```
+
+**Get Single Member API**
+
+```csharp
+{ 
+  "id": "<<GUID>>", 
+  "name": "AnonTest (Guest)", 
+  "tenantId": "<<GUID>>", 
+  "userRole": "anonymous" 
+} 
+```
+
```

Comment:
@v-sdhakshina  lets add a table 
|Property Name|Description|
|---|---|
| id| 29:XXXX if bot supports pairwise version, otherwise 8:teamvisitor:guid  |
| name | provided by anonymous user, when joining meeting |
| tenantId| tenant id of the meeting organizer |
| userRole| `anonymous`, depicting anonymous user|
| userLicenseType | 'Unknown' value depicting anonymous user|
|loginHint| Invalid value, loginHint is not supported for anonymous user |
|userPrincipalName| Invalid value, userPrincipalName is not supported for anonymous user |
|tid| tenant id of the meeting organizer|

Reply:
@v-sdhakshina add a note here
The id returned in Bot api payload and Teams Client SDK are not same.